### PR TITLE
Update scabbard to handle both json lists and csv for arguments

### PR DESCRIPTION
**Pull request by agunde406**
This PR also updates the `splinter circuit propose   --service-arg` to take the value as provided, before it was wrapped in a Vec but this caused issues with setting the version service arg because that should be in a vec. 

The Scabbard factory can now handle peer_services and admin_keys in either json fmt or csv:

`--service-arg "*::admin_keys=admin_keys=key1,key2"`
or 
`--service-arg "*::admin_keys=[\"key1\", \"key2\"]"`

### CODE REVIEW

File: services/scabbard/libscabbard/src/service/factory.rs [hunk 1]
```diff
@@ -156,15 +156,25 @@ impl ServiceArgValidator for ScabbardArgValidator {
...
+        };
 
         for key in admin_keys {
-            let key_bytes = parse_hex(&key).map_err(|_| {
+            let key_bytes = parse_hex(&key).map_err(|err| {
                 ServiceArgValidationError(format!(
-                    "{} is not a valid hex-formatted public key",
-                    key
+                    "{:?} is not a valid hex-formatted public key {}",
+                    key, err
```
Comment (vaporos) [inline services/scabbard/libscabbard/src/service/factory.rs:177]: Is err relevant here? If it is, should there be a : before {}?

**Review by vaporos — APPROVED** [review]

File: services/scabbard/libscabbard/src/service/factory.rs [hunk 2]
```diff
@@ -156,15 +156,25 @@ impl ServiceArgValidator for ScabbardArgValidator {
         let admin_keys_str = args
             .get("admin_keys")
             .ok_or_else(|| ServiceArgValidationError("admin_keys argument not provided".into()))?;
-        let admin_keys: Vec<String> = serde_json::from_str(admin_keys_str).map_err(|err| {
```
Comment (peterschwarz) [inline services/scabbard/libscabbard/src/service/factory.rs:159]: Typo in commit message: s/facotry/factory

File: services/scabbard/libscabbard/src/service/factory.rs (hunk 1 above)
Comment (agunde406) [reply services/scabbard/libscabbard/src/service/factory.rs:177]: I actually meant to remove this. I added it when I was debugging. I will remove it.

File: services/scabbard/libscabbard/src/service/factory.rs [hunk 3]
```diff
@@ -156,15 +156,25 @@ impl ServiceArgValidator for ScabbardArgValidator {
...
-            ServiceArgValidationError(format!("failed to parse admin_keys list: {}", err,))
-        })?;
+
+        let admin_keys = {
+            if admin_keys_str.starts_with('[') {
```
Comment (peterschwarz) [inline services/scabbard/libscabbard/src/service/factory.rs:161]: Can we pull this list parsing out to a function? (you do the same operation three times here)

**Review by peterschwarz — CHANGES_REQUESTED** [review]

File: services/scabbard/libscabbard/src/service/factory.rs (hunk 1 above)
Comment (agunde406) [reply services/scabbard/libscabbard/src/service/factory.rs]: fixed

File: services/scabbard/libscabbard/src/service/factory.rs (hunk 2 above)
Comment (agunde406) [reply services/scabbard/libscabbard/src/service/factory.rs:159]: fixed

File: services/scabbard/libscabbard/src/service/factory.rs (hunk 3 above)
Comment (agunde406) [reply services/scabbard/libscabbard/src/service/factory.rs]: added

**Review by peterschwarz — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Review by peterschwarz — APPROVED** [review]

**Pull request merged**